### PR TITLE
Zun Capsule Creation Method

### PR DESCRIPTION
**Pull request by kevin-zhaoshuai**
Prior to a PR being reviewed, there needs to be a Github issue that the PR
addresses. Replace the brackets and text below with that issue number.

For #700 

Depend-on PR: #944

### CODE REVIEW

File: openstack/container/v1/capsules/requests.go [hunk 1]
```diff
@@ -4,10 +4,53 @@ import (
...
+
+// Create implements create capsule request.
+func Create(client *gophercloud.ServiceClient, opts CreateOptsBuilder) (r CreateResult) {
+	b, err := opts.ToCapsuleCreateMap()
+	if err != nil {
+		r.Err = err
+		return r
+	}
+	_, r.Err = client.Post(createURL(client), b, &r.Body, &gophercloud.RequestOpts{OkCodes: []int{202}})
+	return
```
Comment (hongbin) [inline openstack/container/v1/capsules/requests.go]: I guess here also needs to return the CreateResult?

Comment (jtopjian) [reply openstack/container/v1/capsules/requests.go]: It should be returned automatically. Actually, line 52 should just read `return`.

Go has a feature called "named returns". This explains it better than I could :)

https://tour.golang.org/basics/7

Comment (hongbin) [reply openstack/container/v1/capsules/requests.go]: I see. Glad to learn something news :)

**Comment by hongbin** [comment]
LGTM

**Comment by jtopjian** [comment]
Just a quick note that when this is ready for an official review, please leave a comment like "ready for review" or similar. Otherwise we'll still consider this a work in progress or simply lose track of it in the other PRs.

Also: are the tests running in OpenLab? It's OK if they aren't -- I'll just have to get my zun environment running again to verify.

**Comment by kevin-zhaoshuai** [comment]
@hongbin @jtopjian  Thanks for review. I will address the comments in #858  Ready for review now

File: openstack/container/v1/capsules/errors.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
+package capsules
+
+import (
+	"fmt"
+
+	"github.com/gophercloud/gophercloud"
+)
+
+type ErrInvalidEnvironment struct {
```
[the remark this answers is not in the document]
Comment (kevin-zhaoshuai) [reply openstack/container/v1/capsules/errors.go]: Done

File: openstack/container/v1/capsules/errors.go [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+type ErrInvalidDataFormat struct {
+	gophercloud.BaseError
+}
+
+func (e ErrInvalidDataFormat) Error() string {
+	return fmt.Sprintf("Data in neither json nor yaml format.")
+}
+
+type ErrInvalidTemplateFormatVersion struct {
```
[the remark this answers is not in the document]
Comment (kevin-zhaoshuai) [reply openstack/container/v1/capsules/errors.go]: Done

**Comment by jtopjian** [comment]
@kevin-zhaoshuai sounds good!

The last thing remaining is the `openstack/orchestration/v1/stacks/template.go` file. If you click on the "Files Changed" tab at the top of this page and scroll to the bottom, you'll see that this file is still part of these commits / PR. Would you be able to remove that?

**Comment by kevin-zhaoshuai** [comment]
@jtopjian 
Sorry about that mistake.
Now removed it.

**Comment by jtopjian** [comment]
Looks good to me! Thank you for your work on this :)

**Pull request merged**